lightning/src/ln/channelmanager.rs

```diff
@@ -8840,17 +8849,15 @@ macro_rules! create_offer_builder { ($self: ident, $builder: ty) => {
 
 		let nonce = Nonce::from_entropy_source(entropy);
 		let context = OffersContext::InvoiceRequest { nonce };
-		let path = $self.create_blinded_paths_using_absolute_expiry(context, absolute_expiry)
+		let mut builder = OfferBuilder::deriving_signing_pubkey(node_id, expanded_key, nonce, secp_ctx)
+			.chain_hash($self.chain_hash);
+		if let Some(params) = params {
+			let path = $self.create_blinded_paths(params, context)
 				.and_then(|paths| paths.into_iter().next().ok_or(()))
 				.map_err(|_| Bolt12SemanticError::MissingPaths)?;
-		let builder = OfferBuilder::deriving_signing_pubkey(node_id, expanded_key, nonce, secp_ctx)
-			.chain_hash($self.chain_hash)
-			.path(path);
-		
-		let builder = match absolute_expiry {
-			None => builder,
-			Some(absolute_expiry) => builder.absolute_expiry(absolute_expiry),
-		};
+
+			builder = builder.path(path);
+		}
```

Comment:
Could we use the `match` syntax instead? That way our variables don't need to be `mut`.